python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -167,6 +169,49 @@ def decorated(self, *args, **kwargs):
     return decorated
 
 
+class RegistryFlusher(threading.Thread):
+    """
+    This class is intended to be used by the registry to perfom
+    automatic flushes on a fixed schedule so that information is not
+    lost if Ganga is shut down abruptly.
+    """
+    def __init__(self, registry, *args, **kwargs):
+        super(RegistryFlusher, self).__init__(*args, **kwargs)
+        self.registry = registry
+        self._stop = threading.Event()
+
+    def stop(self):
+        """
+        Ask the thread to stop what it is doing and it will finish
+        the next chance it gets.
+        """
+        self._stop.set()
+
+    @property
+    def stopped(self):
+        return self._stop.isSet()
+
+    def join(self, *args, **kwargs):
+        self.stop()
+        super(RegistryFlusher, self).join(*args, **kwargs)
+
+    def run(self):
+        """
+        This will run an indefinite loop which periodically checks
+        whether it should stop. In between calls to ``flush_all`` it
+        will wait for a fixed period of time.
+        """
+        sleep_period = 30
```

Comment:
Can we have this configurable at all?
This would be quite useful to adjust when/if there are problems and I would like to access it for a unittest I'm currently writing